Tutorial: Framer analytics (#6529)
PostHog/posthog.com: contents/tutorials/framer-analytics.md

---

> title: How to set up Framer analytics, session replay, and more

Reviewer: Should we call it session recording? I think this is more likely to be searched for than session replay

```suggestion
title: How to set up Framer analytics, session recording, and more
```

Author: I think they are about the same, specifically for Framer, it doesn't seem like people are searching either yet, so I'll just keep replay to stay internally consistent.

date: 2023-08-17
author: ["ian-vanagas"]
showTitle: true
sidebar: Docs
featuredImage: ../images/tutorials/banners/tutorial-7.png
tags: []
---

[Framer](https://www.framer.com/) is a popular no-code site builder that makes it easy to design a high-quality site. 

> To maximize the effectiveness of your Framer site, you need analytics about how users are using it. Data like pageviews, button clicks, and session replays are critical to doing to improvement. 

Reviewer: ```suggestion
To maximize the effectiveness of your Framer site, you need analytics to see how users are using it. Data like page views, button clicks, and session replays are critical to improving your site. 
```

> This tutorial shows you how to set up PostHog on your Framer site for event autocapture, session replays, custom event capture, and feature flags.

Reviewer: ```suggestion
This tutorial shows you how to set up PostHog on your Framer site for capturing events, session replays, and implementing feature flags.
```

    posthog.init('<ph_project_api_key>',{api_host:'<ph_instance_address>'})
</script>
```

With the JavaScript snippet copied, go to your Framer project and click the gear in the top right to go to your site settings. If you haven’t already, sign up for the "Mini" site plan. This enables you to add custom code.

Once on a paid plan, go to the "General" tab in site settings and scroll down to the "Custom Code" section. Under "End of `<head>` tag", paste your PostHog JavaScript snippet here. Make sure to press "Save" next to the "Custom Code" heading. 

> ![Script](../images/tutorials/framer-analytics/script.png)

Reviewer: video could be nice here

Author: got it 👍

> Once you publish and go to your site, PostHog starts to autocapture events like pageviews, button clicks, and inputs for your site.

Reviewer: ```suggestion
Once you publish and go to your site, PostHog begins automatically capturing events like page views, button clicks, and form inputs on your site.
```

![Autocapture](../images/tutorials/framer-analytics/autocapture.png)

### Enabling session replays

To enable session replays, go to [your project settings](https://app.posthog.com/project/settings#recordings), scroll down to "Recordings," and enable "Record user sessions." PostHog then starts to record sessions of your users, so you can see all the details of how they use your site.

## Adding a custom event

> You can add [custom code components](https://www.framer.com/developers/#code-components) to Framer and because of that, we can set up [custom event capture](/docs/libraries/js#capturing-events). We will create a basic button that captures a `clicked homepage button`  and sets a `clicked_homepage_button` person property to `true`.

Reviewer: ```suggestion
We'll use [custom code components](https://www.framer.com/developers/#code-components) in Framer to set up [custom event capture](/docs/libraries/js#capturing-events). We will create a basic button that captures a `clicked homepage button`  and sets a `clicked_homepage_button` person property to `true`.
```

> To do this, go to the "Assets" tab in the top left of your project. Next to the "Code" tab, click the plus icon to create a code file. Name the file `CaptureButton`, set it as a "New component" and press "Create."

Reviewer: video or screenshot could be nice

Author: It's included in the video below so I'll just leave as is.

> Once done, delete the boilerplate code and add a button that captures an event with `window.posthog()` like this:

Reviewer: ```suggestion
Once done, delete the existing code and add a button that captures an event with `window.posthog()` like this:
```

```js
export default function CaptureButton() {
    const handleClick = () => {
        window.posthog.capture("clicked homepage button", {
            $set_once: { clicked_homepage_button: true },
        })
    }

    return <button onClick={handleClick}>Click me</button>
}
```

After pasting this in, press cmd/ctrl + S to save the component, and go back to the home page by clicking "Home" above the code. Once on the home page, drag and drop the `CaptureButton` component from the sidebar into your page and publish again to update your site with your new button.

![Framer button video](../images/tutorials/framer-analytics/framer-button.mp4)

Once updated, you can head to your site and click the button to see events captured in PostHog and person properties set on the user.

![Custom event and person properties](../images/tutorials/framer-analytics/person.png)

## Customizing components with feature flags

> Similarly, we can use [feature flags](/docs/feature-flags) in our custom code components. To do this, first, go to the [feature flag tab](https://app.posthog.com/feature_flags) in PostHog. Click "New feature flag," enter a key (I choose `framer-example`), fill out the details, set release conditions to roll out to 100% of users, and click "Save."

Reviewer: Perhaps we should give one line sentence of what feature flags do? or why you'd want to add them
